homeassistant/components/renault/sensor.py

```diff
@@ -45,224 +195,108 @@ async def async_setup_entry(
-    if "cockpit" in vehicle.coordinators:
-        entities.append(RenaultMileageSensor(vehicle, "Mileage"))
-        if vehicle.details.uses_fuel():
-            entities.append(RenaultFuelAutonomySensor(vehicle, "Fuel Autonomy"))
-            entities.append(RenaultFuelQuantitySensor(vehicle, "Fuel Quantity"))
-    if "hvac_status" in vehicle.coordinators:
-        entities.append(RenaultOutsideTemperatureSensor(vehicle, "Outside Temperature"))
-    if "battery" in vehicle.coordinators:
-        entities.append(RenaultBatteryLevelSensor(vehicle, "Battery Level"))
-        entities.append(RenaultChargeStateSensor(vehicle, "Charge State"))
-        entities.append(
-            RenaultChargingRemainingTimeSensor(vehicle, "Charging Remaining Time")
-        )
-        entities.append(RenaultChargingPowerSensor(vehicle, "Charging Power"))
-        entities.append(RenaultPlugStateSensor(vehicle, "Plug State"))
-        entities.append(RenaultBatteryAutonomySensor(vehicle, "Battery Autonomy"))
-        entities.append(
-            RenaultBatteryAvailableEnergySensor(vehicle, "Battery Available Energy")
-        )
-        entities.append(RenaultBatteryTemperatureSensor(vehicle, "Battery Temperature"))
-    if "charge_mode" in vehicle.coordinators:
-        entities.append(RenaultChargeModeSensor(vehicle, "Charge Mode"))
-    return entities
-
-
-class RenaultBatteryAutonomySensor(RenaultBatteryDataEntity, SensorEntity):
-    """Battery autonomy sensor."""
-
-    _attr_icon = "mdi:ev-station"
-    _attr_native_unit_of_measurement = LENGTH_KILOMETERS
-
-    @property
-    def native_value(self) -> int | None:
-        """Return the state of this entity."""
-        return self.data.batteryAutonomy if self.data else None
-
-
-class RenaultBatteryAvailableEnergySensor(RenaultBatteryDataEntity, SensorEntity):
-    """Battery available energy sensor."""
-
-    _attr_device_class = DEVICE_CLASS_ENERGY
-    _attr_native_unit_of_measurement = ENERGY_KILO_WATT_HOUR
-
-    @property
-    def native_value(self) -> float | None:
-        """Return the state of this entity."""
-        return self.data.batteryAvailableEnergy if self.data else None
-
-
-class RenaultBatteryLevelSensor(RenaultBatteryDataEntity, SensorEntity):
-    """Battery Level sensor."""
-
-    _attr_device_class = DEVICE_CLASS_BATTERY
-    _attr_native_unit_of_measurement = PERCENTAGE
-
-    @property
-    def native_value(self) -> int | None:
-        """Return the state of this entity."""
-        return self.data.batteryLevel if self.data else None
+        for description in SENSOR_TYPES:
+            if description.coordinator in vehicle.coordinators:
+                create_entity = True
+                if description.requires_fuel:
+                    create_entity = vehicle.details.uses_fuel()
+                if create_entity:
+                    entity_class = globals()[description.entity_class]
+                    entities.append(entity_class(vehicle, description))
```

Comment:
Use a list comprehension for `entities`.

Reply:
This should be OK now